+ (Lava) Fixed issue in Group Registration block where lava variable was always empty list. (Fixes #5224)

courtneycooksey opened the pull request:
## Notice

**We cannot guarantee that your pull request will be accepted.**  There are many factors involved.  We take into consideration whether or not the Rock system you run is a standard, main-line build.  If it is not, there is a lower chance we will accept your request (since it may impact a part of the system you don't regularly use).  Features that would be used by less than 80% of Rock organizations, or ones that don't match the goals of Rock, are other important factors.

## Proposed Changes

Pass groupMembers variable with ref so when it is updated in the `AddPersonToGroup` function it updates the original value that is later added to the lava context. 
Add newly created group member entities to the groupMembers list. 

Fixes: #5224

## Types of changes

What types of changes does your code introduce to Rock?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality, which has been approved by the core team)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] This is a single-commit PR. (If not, please squash your commit and re-submit it.)
- [x] I verified my PR does not include whitespace/formatting changes -- because if it does it will be closed without merging.	
- [x] I have read the [Contributing to Rock](https://github.com/SparkDevNetwork/Rock/blob/master/.github/CONTRIBUTING.md) doc
- [x] By contributing code, I agree to license my contribution under the [Rock Community License Agreement](https://www.rockrms.com/license)
- [x] Unit tests pass locally with my changes
- [x] I have added any required [unit tests](https://github.com/SparkDevNetwork/Rock/blob/develop/Rock.Tests.UnitTests/README.md) or [integration tests](https://github.com/SparkDevNetwork/Rock/blob/develop/Rock.Tests.Integration/README.md) that prove my fix is effective or that my feature works
- [x] I have included updated language for the [Rock Documentation](https://www.rockrms.com/Learn/Documentation) (if appropriate)

## Further comments
Small change, I ran the hotkey for formatting on the block as well.

MrUpsideDown reviewed (commented):
@courtneycooksey  - thanks for taking the time to prepare this fix! From what I can see, I don't think the "ref" keyword  is needed for the AddPersonToGroup "groupMembers" parameter, because the collection being referenced doesn't change (although the contents of the collection itself are modified). Your thoughts?

courtneycooksey commented:
You're right. It wasn't working when I was first messing with it, so I added that. I'm pretty sure I remember having messed up the lava variable name though, so I probably just didn't re-test after I fixed my typo. Do you want me to close this and submit a new pr?

MrUpsideDown commented:
I think it's OK if you just amend this PR with the changes?
